terrascan.sh

```diff
@@ -4,27 +4,42 @@ set -eo pipefail
 main() {
   initialize_
   parse_cmdline_ "$@"
-
-  # propagate $FILES to custom function
-  terrascan_ "$ARGS" "$FILES"
+  terrascan_
 }
 
 terrascan_() {
   # consume modified files passed from pre-commit so that
   # terrascan runs against only those relevant directories
-  for file_with_path in $FILES; do
+  for file_with_path in "${FILES[@]}"; do
     file_with_path="${file_with_path// /__REPLACED__SPACE__}"
     paths[index]=$(dirname "$file_with_path")
-
-    let "index+=1"
+    ((index++)) || true
```

Comment:
`|| true` is a bit non-obvious hack to work around how the `((expression))` works alongside `errexit`, hence I'd suggest to use more straightforward and `errexit`-independent approach:
```suggestion
    index=$((index+1))
```
Or just leave this line as is (`    let "index+=1"`).

Reply:
@yermulnik Thanks for pointing that, is good to know!